04/app.js

```diff
@@ -31,10 +35,34 @@ function getChannelColor(colorInHex, channelName) {
             break;
     }
 
-    const channelColorHex = colorInHex.substr(start, 2);
+    const channelColorHex = colorInHex.substr(start, 2); // substr(start,lenght) - Zwraca określoną liczbę początkowych znaków w łańcuchu znaków w określonej lokalizacji.
     const channelColorDec = parseInt(channelColorHex, 16);
 
-    return channelColorDec; 
+    return channelColorDec;
 }
 
+//Działa cześciowo, nie moge zmienić shadow na kolorze początkowym i wydaje mi sie ze to rozwiązanie mozna lepiej napsiac ale nie mam juz pomysłu
+pannelEl.addEventListener('change', function(e){
+    let pickedRange;
+    if(e.target.type ==='color'){
+        pickedColor = e.target.value;
+    }
+    if(e.target.type ==='range'){
+        pickedRange = Number(e.target.value)/100;
+    }
+    setBoxShadow(boxElement,pickedColor,pickedRange);
+});
+
+
+// inputColor.addEventListener('change', function(e){
+//     pickedColor = e.target.value;
+//     setBoxShadow(boxElement,pickedColor);
+// });
```

Comment:
To jest odpowiedni kierunek tj. osobne zdarzenia na każdy z elementów, ale... trzeba dodać też wartość kolejną czyli np. `setBoxShadow(boxElement,pickedColor, inputRange.value);`